### DISCUSSION TITLE
Documentation of Certificate related API's

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Lern/Community#112] [Documentation] [answered]
We went through few documentations of Cert by taking Lern BB team help. Still we are struggling to setup the Cert to workflow to enable. 

**Document**: [Certificate Configuration Implementation](https://docs.google.com/document/d/1bAKi8GFX4u1iZvIMHMef9P76OaDs_wVb-jK0CShpmeU/edit) 
**Micosite**: https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/api-documentation#course-batch-cert-v1-template-add
**Discussion Threads:**
[Not getting Certificate templates for AP and NCERT tenants in Staging env · Sunbird-Lern · Discussion #47](https://github.com/orgs/Sunbird-lern/discussions/47) 

**Design wikis**:
[Certificate: Attach to a batch | Create: Cert Template for a tenant](https://project-sunbird.atlassian.net/wiki/spaces/SBDES/pages/1583677441/Certificate+Attach+to+a+batch#Create:-Cert-Template-for-a-tenant)

It will be useful for any adopter if the team document all the API's related to certificate & share the sequence of the API's to be executed to setup the Cert workflow.

Suggestion of sequence of API's:
1. Uploading default certificate templates.
   Using these default template tenant can create tenant specific certificate templates by uploading logo's & signatures.
2. Creation of tenant specific templates(these can to attached to courses)
3. Form API's that are used in creation of certificate templates(score, completion criteria, Logo & signatures etc)
4. Attaching Certificate to the Course.
5. Issue/re-issue of certificate to the User based on criteria.

We have noticed still there are some references from Microsite pointing to old Sunbird API documentation    (http://docs.sunbird.org/latest/apis/coursebatchcertificateapi/)

@Sunbird-Lern/sunbirdlern_workinggroup [@Krishnaj20](https://github.com/Krishnaj20) [@rajeevsathish](https://github.com/rajeevsathish) @gandham-santhosh

**Comment by rhwarrier** [comment]
Thanks for the inputs @vinukumar-vs - this will help us structure our documentation better. The team is in the process of improving the documentation, as well as identifying and filling gaps that exist. 
@reshmi-nair  , @vinukumar-vs , @BharathwajShankar  please work closely with the ED team to understand where they're facing issues - this will also help us get the microsite to a comprehensive state quickly.

**Reply by Jayaprakash8887** [reply]
@vinukumar-vs Though major part of the Functional flow is part of LERN, some parts of the same is part of KNOWLG. As the integration of the same is done by ED, flow implementation documentation can be done by ED maybe. For starters, you can probably refer to this old documentation prepared by me as part of implementation team: https://tarento-my.sharepoint.com/:w:/p/jayaprakash_narayanaswamy/Efof_5XKidxZymTGVKc8aHcBx5o8GVYN-FQwrx4PAqtBfA

**Reply by Jayaprakash8887** [reply]
For all recent releases, tenants need to follow steps for configuring a “cert template layout” only as they can generate certificate templates now in the UI flow as course creator while creating batch.

**Deprecated**: Steps for configuring a “cert template”

**Reply by vinukumar-vs** [reply]
@Jayaprakash8887 I would be better to put all the documents at place(Gitbook). From Githook you give reference link to any external documents like Google docs, Postman Collection, Confluence wiki's etc..
This will help for adopters to find at single place for any documentation reference. 

As ED we used Postman collection for API's(function flow API's & other service API's owned by ED). All these collections are linked to Gitbook for ease of discovery for adopters(still need to improve)
https://ed.sunbird.org/use/apis

@reshmi-nair @rhwarrier @BharathwajShankar

**Reply by Krishnaj20** [reply]
@rhwarrier - is this prioritized by the lern team?

**Reply by rhwarrier** [reply]
Yes, we're closing on all documentation related issues as part of the current release
@Jayaprakash8887 @BharathwajShankar @AmiableAnil pls ensure we pick up all such issues raised by adopters and close out all gaps.

**Reply by vinod303** [reply]
Done[ this document ](https://drive.google.com/file/d/1NI5nsxsgcro2wMZmW__JZUrUmFLUTCSY/view?usp=sharing
)which explains step by step how to create and configure Certificate in the system. Please review and if this requires some changes / formatting or add more details , happy to do that.

https://drive.google.com/file/d/1NI5nsxsgcro2wMZmW__JZUrUmFLUTCSY/view?usp=sharing

@Krishnaj20 @rhwarrier @reshmi-nair @vinukumar-vs

**Reply by rhwarrier** [reply]
Thank you, @vinod303  !  @AmiableAnil @BharathwajShankar  pls review, and use this for the microsite as well.

**Reply by AmiableAnil** [reply]
Sure @rhwarrier

**Reply by Jayaprakash8887** [reply, marked as answer]
@vinukumar-vs please refer to https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/certificate-flow

Kindly mark this thread as answered. Also, please note that we will work on this documentation further in future.
